File: app/src/main/java/com/digicoffer/lauditor/Documents/DocumentsListAdpater/GroupsListAdapter.java
package com.digicoffer.lauditor.Documents.DocumentsListAdpater;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.digicoffer.lauditor.Documents.Documents;
import com.digicoffer.lauditor.Documents.Models.DocumentsModel;
import com.digicoffer.lauditor.R;

import java.util.ArrayList;

public class GroupsListAdapter extends RecyclerView.Adapter<GroupsListAdapter.Viewholder> {
    private ArrayList<DocumentsModel> sharedList = new ArrayList<>();
    private ArrayList<DocumentsModel> list_item = new ArrayList<>();
    private OnCheckedChangeListener listener;

    public GroupsListAdapter(ArrayList<DocumentsModel> sharedList, Documents documents, OnCheckedChangeListener listener) {
        this.sharedList = sharedList;
        this.list_item = sharedList;
        this.listener = listener;
    }

    @NonNull
    @Override
    public Viewholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.select_team_members, parent, false);
        return new Viewholder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull Viewholder holder, int position) {
        DocumentsModel groupsModel = sharedList.get(position);
        holder.cb_documents.setChecked(sharedList.get(position).isGroupChecked());
        holder.cb_documents.setTag(position);
        holder.tv_tm_name.setText(groupsModel.getGroup_name());

        // Setting click listener for the checkbox
        holder.cb_documents.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Integer pos = (Integer) holder.cb_documents.getTag();
                if (sharedList.get(pos).isGroupChecked()) {
                    sharedList.get(pos).setGroupChecked(false);
                    holder.cb_documents.setChecked(false);
                } else {
                    sharedList.get(pos).setGroupChecked(true);
                    holder.cb_documents.setChecked(true);
                }


                if (listener != null) {
                    listener.onCheckedChanged(sharedList.get(pos));
                }
            }
        });
    }

    public ArrayList<DocumentsModel> getList_item() {
        return sharedList;
    }

    @Override
    public int getItemCount() {
        return sharedList.size();
    }

    public class Viewholder extends RecyclerView.ViewHolder {
        private TextView tv_tm_name;
        private View list_line;
        private LinearLayout select_tm_layout;
        private CheckBox cb_documents;

        public Viewholder(@NonNull View itemView) {
            super(itemView);
            cb_documents = itemView.findViewById(R.id.chk_selected);
            tv_tm_name = itemView.findViewById(R.id.tv_tm_name);
            select_tm_layout = itemView.findViewById(R.id.select_tm_layout);
            list_line = itemView.findViewById(R.id.list_line);
            list_line.setVisibility(View.VISIBLE);
//            LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
//            params.setMargins(0, 0, 0, 0);
//            select_tm_layout.setLayoutParams(params);
            select_tm_layout.setBackgroundResource(R.drawable.background_transparent);
        }
    }

    // Interface for checkbox change listener
    public interface OnCheckedChangeListener {
        void onCheckedChanged(DocumentsModel documentsModel);
    }
}

File: app/src/main/java/com/digicoffer/lauditor/Documents/DocumentsListAdpater/GroupsListAdapter.kt
package com.digicoffer.lauditor.Documents.DocumentsListAdpater

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.LinearLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.digicoffer.lauditor.Documents.Documents
import com.digicoffer.lauditor.Documents.Models.DocumentsModel
import com.digicoffer.lauditor.R
import java.util.ArrayList

class GroupsListAdapter(
    private var sharedList: ArrayList<DocumentsModel>,
    documents: Documents?,
    private val listener: OnCheckedChangeListener?
) : RecyclerView.Adapter<GroupsListAdapter.Viewholder>() {

    private var list_item: ArrayList<DocumentsModel> = sharedList

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Viewholder {
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.select_team_members, parent, false)
        return Viewholder(itemView)
    }

    override fun onBindViewHolder(holder: Viewholder, position: Int) {
        val groupsModel = sharedList[position]
        holder.cb_documents.isChecked = groupsModel.isGroupChecked
        holder.cb_documents.tag = position
        holder.tv_tm_name.text = groupsModel.group_name

        // Setting click listener for the checkbox
        holder.cb_documents.setOnClickListener {
            val pos = holder.cb_documents.tag as Int
            if (sharedList[pos].isGroupChecked) {
                sharedList[pos].isGroupChecked = false
                holder.cb_documents.isChecked = false
            } else {
                sharedList[pos].isGroupChecked = true
                holder.cb_documents.isChecked = true
            }

            listener?.onCheckedChanged(sharedList[pos])
        }
    }

    fun getList_item(): ArrayList<DocumentsModel> {
        return sharedList
    }

    override fun getItemCount(): Int {
        return sharedList.size
    }

    inner class Viewholder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tv_tm_name: TextView = itemView.findViewById(R.id.tv_tm_name)
        val list_line: View = itemView.findViewById(R.id.list_line)
        val select_tm_layout: LinearLayout = itemView.findViewById(R.id.select_tm_layout)
        val cb_documents: CheckBox = itemView.findViewById(R.id.chk_selected)

        init {
            list_line.visibility = View.VISIBLE
            select_tm_layout.setBackgroundResource(R.drawable.background_transparent)
        }
    }

    // Interface for checkbox change listener
    fun interface OnCheckedChangeListener {
        fun onCheckedChanged(documentsModel: DocumentsModel)
    }
}
